target/linux/ath79/dts/qca9558_tplink_archer-c.dtsi

```diff
@@ -115,36 +116,189 @@
 &mdio0 {
 	status = "okay";
 
-	phy0: ethernet-phy@0 {
+	switch@0 {
+		compatible = "qca,qca8327";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
 		reg = <0>;
 
-		qca,ar8327-initvals = <
-			0x04 0x00080080 /* PORT0 PAD MODE CTRL */
-			0x0c 0x07600000 /* PORT6 PAD MODE CTRL */
-			0x50 0xc737c737 /* LED_CTRL0 */
-			0x54 0x00000000 /* LED_CTRL1 */
-			0x58 0x00000000 /* LED_CTRL2 */
-			0x5c 0x0030c300 /* LED_CTRL3 */
-			0x7c 0x0000007e /* PORT0_STATUS */
-			0x94 0x0000007e /* PORT6 STATUS */
-			>;
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			port@0 {
+				reg = <0>;
+				label = "cpu";
+				ethernet = <&eth0>;
+				phy-mode = "rgmii-id";
```

Comment:
original initvals specify port0 sgmii, port 6 rgmii-id.

Reply:
Yep you are correct. I'm testing different configs if they make any difference. 0x07600000 actually means RGMII with 1ns tx and 2ns rx delays, but those are also defaults in qca8k if delays are not set.

It may be that only port 0 is used for Ethernet MDIO. When you enable only the RGMII port connected to port 6, I don't see any extra Ethernet frames. Also all ports work without problems. I need to debug this a bit more when MDIO over Ethernet is actually enabled.

Reply:
According to @Ansuel, using external MDIO or not using SGMII disables mgmt eth. This is a work around. It does not fix the fundamental issue. It should also be slower.